### ISSUE TITLE
Wrong size Error while running basset_train.lua with my own dataset

### ISSUE THREAD

**Issue by kimmo1019** [davek44/Basset#30]
Hi Dave,
I followed the instructions mentioned in the tutorial file: new_data_iso.ipynb
in my dataset, I hava only only one positive bed file and one negative bed file , after preprocess I got:
68534 training sequences 
6000 test sequences 
6000 validation sequences
saved in learn_cd4.h5. (length equals 600 as you suggested)
then I created a blank params.txt to save training parameters.
In the last step I try to run "basset_train.lua -job params.txt -save cd4_cnn learn_cd4.h5" 

## I got the following error:
[liuqiao@g01 mydata]$ basset_train.lua -job params.txt -save cd4_cnn learn_cd4.h5
{}
seq_len: 600, filter_size: 10, pad_width: 9	
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> (9) -> output]
  (1): nn.SpatialConvolution(4 -> 10, 10x1, 1,1, 4,0)
  (2): nn.SpatialBatchNormalization (4D) (10)
  (3): nn.ReLU
  (4): nn.Reshape(6000)
  (5): nn.Linear(6000 -> 500)
  (6): nn.BatchNormalization (2D) (500)
  (7): nn.ReLU
  (8): nn.Linear(500 -> 1)
  (9): nn.Sigmoid
}
/home/liuqiao/torch/install/bin/luajit: /home/liuqiao/torch/install/share/lua/5.1/nn/Container.lua:67: 
In 4 module of nn.Sequential:
/home/liuqiao/torch/install/share/lua/5.1/torch/Tensor.lua:466: **Wrong size for view. Input size: 128x10x1x599. Output size: 128x6000**
stack traceback:
	[C]: in function 'error'
	/home/liuqiao/torch/install/share/lua/5.1/torch/Tensor.lua:466: in function 'view'
	/home/liuqiao/torch/install/share/lua/5.1/nn/Reshape.lua:46: in function </home/liuqiao/torch/install/share/lua/5.1/nn/Reshape.lua:31>
	[C]: in function 'xpcall'
	/home/liuqiao/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
	/home/liuqiao/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function 'forward'
	/home/liuqiao/Basset/src/convnet.lua:1030: in function 'opfunc'
	/home/liuqiao/torch/install/share/lua/5.1/optim/rmsprop.lua:35: in function 'rmsprop'
	/home/liuqiao/Basset/src/convnet.lua:1063: in function 'train_epoch'
	/home/liuqiao/Basset/src/basset_train.lua:148: in main chunk
	[C]: in function 'dofile'
	...qiao/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x004064f0

WARNING: If you see a stack trace below, it doesn't point to the place where this error occurred. Please use only the one above.
stack traceback:
	[C]: in function 'error'
	/home/liuqiao/torch/install/share/lua/5.1/nn/Container.lua:67: in function 'rethrowErrors'
	/home/liuqiao/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function 'forward'
	/home/liuqiao/Basset/src/convnet.lua:1030: in function 'opfunc'
	/home/liuqiao/torch/install/share/lua/5.1/optim/rmsprop.lua:35: in function 'rmsprop'
	/home/liuqiao/Basset/src/convnet.lua:1063: in function 'train_epoch'
	/home/liuqiao/Basset/src/basset_train.lua:148: in main chunk
	[C]: in function 'dofile'
	...qiao/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x004064f0

It reported that "Wrong size for view" . I've tried to read your source code in basset_train.lua but I'm still not sure what's wrong with it. I have to say I'm not very familiar with torch and lua. I guess **maybe some command line parameters got wrong or I have to revise some source code?** .I really hope you could help me figure it out or give me some suggestions.
Thanks a lot!

**Comment by kimmo1019** [comment]
I might make a mistake. I use your pretrained_param.txt, then, everything goes well. Besides, I use -cuda and -cudnn to accelerate the training process. 
In the output, I can see the loss and AUC for each epoch

**Epoch #  1   train loss =   0.557, valid loss =   0.561, AUC = 0.7865, time =  41s best!	
Epoch #  2   train loss =   0.516, valid loss =   0.549, AUC = 0.7929, time =  39s best!	
Epoch #  3   train loss =   0.448, valid loss =   0.733, AUC = 0.7635, time =  40s**

I looked into your source code (basset_train.lua)
**acc_str = string.format("AUC = %.4f", valid_acc_avg)
io.write(string.format("valid loss = %7.3f, %s, time = %s", valid_loss, acc_str, time_str))**

I'm quite confused when I noticed that you took the average validation accuracy as the AUC. AUC means the area under (ROC)curve. Am I getting something wrong? Hope you can help me figure it out.
Thanks a lot!

**Comment by davek44** [comment]
You're overfitting. That's likely to happen quickly with the pre-trained model. It has a ton of parameters because it's trained to predict all 164 datasets. If you're refining those parameters for a simpler prediction task, it will immediately overfit. That second epoch model is likely the best you'll see.

The variable has the term "acc" for "accuracy" because I've changed the accuracy measure a few times. At the moment, it's reporting area under the ROC curve. It has the term "avg" for "average" because I compute the accuracy statistic independently for each task and report the average for the user. If you just have one task, then it's taking the average of one element, so it's not really an average, but there's nothing to worry about.
